studio/components/to-be-cleaned/ModalsDeprecated/InviteMemberModal.tsx

```diff
@@ -0,0 +1,145 @@
+import { observer, useLocalObservable } from 'mobx-react-lite'
+import { post } from 'lib/common/fetch'
+import { Button, IconKey, IconMail, Input } from '@supabase/ui'
+import { Modal } from '@supabase/ui'
+
+import { API_URL } from 'lib/constants'
+import { useOrganizationDetail, useStore } from 'hooks'
+import { toJS } from 'mobx'
+
+
+
+/**
+ * Endpoints (delete when finished)
+ *
+ * Delete invite
+ * curl -X DELETE "http://localhost:8080/platform/organizations/:slug/members/invite?invited_id=:id" -H "Authorization: Bearer access_token_jwt"
+ *
+ * Get all pending invites
+ * curl -X GET "http://localhost:8080/platform/organizations/:slug/members/invite" -H "Authorization: Bearer access_token_jwt"
+ *
+ * Send an invitation
+ * curl -X POST "http://localhost:8080/platform/organizations/slug/members/invite" -H "Content-Type: application/json" -d '{"invited_email": "", "owner_id": ""}' -H "Authorization: Bearer ..."
+ *
+ * Verify an invitation
+ * curl -X GET 'http://localhost:8080/platform/organizations/slug/members/join?token=XXX'
+ *
+ */
+
+
+/**
+ * Modal to invite member to Organization
+ *
+ * @param {Object}          organization          // organization detail
+ * @param {Array}           members               // organization members list
+ * @param {Object}          user                  // current user detail
+ */
+
+const PageContext = createContext(null)
+
+function InviteMemberModal({ organization, members = [], user }: any) {
+
+  const PageState = useLocalObservable(() => ({
+    members:  [],
+    addMemberLoading: false,
+    emailAddress: '',
+    emailIsValid() {
+      return String(this.emailAddress)
+        .toLowerCase()
+        .match(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
+    }
+  }))
+  const { ui } = useStore()
+  const { mutateOrgMembers } = useOrganizationDetail(ui.selectedOrganization?.slug || '')
+  const [isOpen, setIsOpen] = useState(false)
+  const { id: orgId, slug: orgSlug } = organization
+
+  useEffect(() => {
+    PageState.members = members
+  }, [members])
+
+  function toggle() {
+    // reset data before showing modal again
+    if (!isOpen) {
+      PageState.addMemberLoading = false
+      PageState.emailAddress = ''
+    }
+    setIsOpen(!isOpen)
+  }
+
+  async function addMember() {
+    PageState.addMemberLoading = true
+
+    const response = await post(`${API_URL}/organizations/${orgSlug}/members/invite`, {
+      invited_email: PageState.emailAddress,
+      owner_id: toJS(user.id)
+    })
+    if (isNil(response)) {
+      ui.setNotification({ category: 'error', message: 'Failed to add member' })
+    } else if (response?.error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to add member: ${response.error.message}`,
+      })
+      PageState.addMemberLoading = false
+    } else {
+      const newMember = response
```

Comment:
Can't do it this way as the `response` from the `/members/invite` endpoint doesn't return the same object structure as members within `PageState.members`

Causes a client error on the `[slug]/settings` page

We could either
- have studio format the response into an object structure which can fit PageState.members OR
- update the response payload of the endpoint itself to return something that the UI can use immediately

Reply:
i've updated the code to the former for now